#!/bin/sh /usr/share/dpatch/dpatch-run
## 12-SecBull-TYPO3-CORE-SA-2011-003.dpatch by Christian Welzel <gawain@camlann.de>
##
## DP: Fixes SecBull-TYPO3-SA-2011-003 (adopted from Core.git
## Change-Id: I01184ebda0f8667b431f371db155f06e9f0ed898)

@DPATCH@

diff --git a/typo3/sysext/cms/tslib/class.tslib_fe.php b/typo3/sysext/cms/tslib/class.tslib_fe.php
index f9448fc..8355180 100644
--- a/typo3/sysext/cms/tslib/class.tslib_fe.php
+++ b/typo3/sysext/cms/tslib/class.tslib_fe.php
@@ -1784,7 +1784,7 @@
 				if ($this->TYPO3_CONF_VARS['FE']['pageNotFoundOnCHashError']) {
 					$this->pageNotFoundAndExit('Request parameters could not be validated (&cHash comparison failed)');
 				} else {
-					$this->set_no_cache();
+					$this->no_cache = true;
 					$GLOBALS['TT']->setTSlogMessage('The incoming cHash "'.$this->cHash.'" and calculated cHash "'.$cHash_calc.'" did not match, so caching was disabled. The fieldlist used was "'.implode(',',array_keys($this->cHash_array)).'"',2);
 				}
 			}
@@ -1804,7 +1804,7 @@
 				if ($this->tempContent)	{ $this->clearPageCacheContent(); }
 				$this->pageNotFoundAndExit('Request parameters could not be validated (&cHash empty)');
 			} else {
-				$this->set_no_cache();
+				$this->no_cache = true;
 				$GLOBALS['TT']->setTSlogMessage('TSFE->reqCHash(): No &cHash parameter was sent for GET vars though required so caching is disabled',2);
 			}
 		}
